File: movies/movies-api-hexa/src/main/java/com/movies/api/domain/model/Movie.java
package com.movies.api.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * Entité métier pure - Aucune dépendance technique
 */
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Movie {
    private Long id;
    private String imdbId;
    private String title;
    private String originalTitle;
    private int releasedDateYear;
    private int runtimeMinute;
    private String story;
    private String imageUri;
    private boolean haveBeenSeen;
    private double ratingValue;
    private Double personalRatingValue;

    // Logique métier pure
    public boolean hasPersonalRating() {
        return personalRatingValue != null;
    }

    public boolean isRecentMovie() {
        return releasedDateYear >= 2020;
    }

    public boolean isLongMovie() {
        return runtimeMinute > 180;
    }
}

File: MoviesApiDotnet/src/MoviesApi.Domain/Entities/Movie.cs
namespace MoviesApi.Domain.Entities;

public class Movie
{
    public long Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Director { get; set; } = string.Empty;
    public int Year { get; set; }
    public double Rating { get; set; }
    public bool HaveBeenSeen { get; set; }
}
